nilearn/plotting/displays.py

```diff
@@ -640,9 +640,12 @@ def _map_show(self, img, type='imshow',
             else:
                 data = np.ma.masked_inside(data, -threshold, threshold,
                                            copy=False)
-            img = new_img_like(img, data, img.get_affine())
+            img = new_img_like(img, data, _utils.compat.get_affine(img))
 
-        affine = img.get_affine()
+        if img.__class__.__name__ == '_MNI152Template':
+            affine = img.get_affine()
+        else:
+            affine = _utils.compat.get_affine(img)
```

Comment:
I don't think so. _MNI152Template has a function ```get_affine``` in plotting module. It returns empty when I tried to call with this PR function. Hence tests are failing completely.